python/vyos/ifconfig/interface.py

```diff
@@ -311,6 +312,28 @@ def _delete(self):
         cmd = 'ip link del dev {ifname}'.format(**self.config)
         return self._cmd(cmd)
 
+    def _set_vrf_ct_zone(self, vrf):
+        """
+        Add/Remove rules in nftables to associate traffic in VRF to an
+        individual conntack zone
+        """
+        conf = Config()
```

Comment:
@dmbaturin is it allowed/wise to instantiate a new `Config()` object here?

It would be easier and no object of type `Config()` would be needed, if the VRF table ID is simply retrieved from the operating system, as VRFs are always created in advance to assigning an interface to a VRF.

```
vyos@vyos:~$ python3
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from vyos.util import get_interface_config
>>> import pprint
>>> pprint.pprint(get_interface_config('red'))
{'address': '62:7b:9c:1b:b1:23',
 'broadcast': 'ff:ff:ff:ff:ff:ff',
 'flags': ['NOARP', 'MASTER', 'UP', 'LOWER_UP'],
 'group': 'default',
 'gso_max_segs': 65535,
 'gso_max_size': 65536,
 'ifindex': 142,
 'ifname': 'red',
 'inet6_addr_gen_mode': 'none',
 'link_type': 'ether',
 'linkinfo': {'info_data': {'table': 2000}, 'info_kind': 'vrf'},
 'linkmode': 'DEFAULT',
 'max_mtu': 65575,
 'min_mtu': 1280,
 'mtu': 65575,
 'num_rx_queues': 1,
 'num_tx_queues': 1,
 'operstate': 'UP',
 'promiscuity': 0,
 'qdisc': 'noqueue',
 'txqlen': 1000}
>>>
```

VRF red has table ID 2000.

Reply:
I think you are right, let me change this.

Reply:
I think retrieving it from the kernel is a better idea indeed.

Creating another `Config` instance _should_ be safe, but retrieving VRFs from the kernel is going to be much faster.

Reply:
Updated.